src/vsr/replica.zig

```diff
@@ -2997,6 +2997,10 @@ pub fn ReplicaType(
         }
 
         fn ignore_prepare_ok(self: *Self, message: *const Message) bool {
+            if (self.primary_index(message.header.view) == self.replica) {
+                assert(message.header.view <= self.view);
```

Comment:
Could this fail if the replica is recovering after a crash (i.e. not in normal status, see below) and then receives a `prepare_ok`?

Reply:
That is exactly the case I am aiming to catch here. If I am recovering, and I receive a prepare_ok **for a view where I am the primary**, and I _don't_ know about the view, something is seriously wrong somewhere. 

This means that I started this "future" view and then forgot about it.

Reply:
(and we have "never forget about views" as our safety invariant after we removed recovery protocol)

Reply:
i.e. Some situation at startup where we do have the latest view on disk, but we're still getting there...

Reply:
> Yes, I mean, what if we are still recovering from the write ahead log?

The other thing to be aware of is that the replica does not process any messages (from the message bus) until it has opened its superblock (recovering its view number) and recovered its whole WAL.